lib/project_types/extension/forms/questions/ask_app.rb

```diff
@@ -0,0 +1,53 @@
+module Extension
+  module Forms
+    module Questions
+      class AskApp
+        include ShopifyCli::MethodObject
+
+        property! :ctx
+        property :api_key
+        property! :prompt,
+          converts: :to_proc,
+          default: -> { CLI::UI::Prompt.method(:ask) }
+
+        def call(project_details)
+          project_details.tap(&method(:prompt_for_app))
+        end
+
+        private
+
+        def prompt_for_app(project_details)
+          project_details.app =
+            api_key.nil? ? choose_interactively(load_apps) : validate_app_ownership(api_key)
+        end
+
+        def validate_app_ownership(api_key)
+          found_app = Tasks::GetApp.call(context: ctx, api_key: api_key)
+          ctx.abort(ctx.message('register.invalid_api_key', api_key)) if found_app.nil?
+          found_app
+        end
+
+        def choose_interactively(apps)
+          prompt.call(ctx.message('register.ask_app')) do |handler|
```

Comment:
Should we move the `ask_app` string under `create.ask_app`?

Reply:
Good catch!